### DISCUSSION TITLE
Initial creation of smart connections takes weeks?

### DISCUSSION THREAD

**Discussion by perryfr** [brianpetro/obsidian-smart-connections#273] [General]
Hello: I'm very excited about using this plug-in. **I'm wondering whether there's a way to speed up the initial process of making smart connections.** I've had my computer on and connected to the internet for close to two weeks since setting up Smart Connections with gpt-3.5-turbo (4k), but at this point OpenAI has used only $0.10 worth of tokens to make smart connections. There was a spike in token usage early on, as you can see in the screenshot below (my sole use of OpenAI during this period has been for Smart Connections). Does anyone know how I can get OpenAI to work as fast as it was working initially? Many thanks for any help you can provide.

<img width="807" alt="CleanShot 2023-06-30 at 11 05 36@2x" src="https://github.com/brianpetro/obsidian-smart-connections/assets/96356433/8d110cd1-8d2f-4c1b-b1d8-5a0a26e40057">

**Comment by brianpetro** [comment]
Hi @perryfr 

Thanks for trying Smart Connections.

It shouldn't take that long.

Are you seeing any errors in the developer console?

Thanks
Brian 🌴

**Reply by perryfr** [reply]
Aha! Yes, the developer panel is bleeding red, as you can see in the screenshot below. 

<img width="542" alt="CleanShot 2023-06-30 at 19 22 15@2x" src="https://github.com/brianpetro/obsidian-smart-connections/assets/96356433/9223d9ce-393b-4161-9a84-96770759f96f">

I should say that early on in the process, I switched from using gpt-3.5-turbo (4k) as my Smart Chat Model to using the 16k one. Since then, when I close and re-open my vault, I get the message in the screenshot below.

![CleanShot 2023-06-30 at 19 28 28@2x](https://github.com/brianpetro/obsidian-smart-connections/assets/96356433/b73b5c11-a4a2-4627-a57b-39288575b2e0)

I assumed everything was fine, though, because whenever I'm in Options for the plug-en and hist the "Test API Key," I get a message saying it's valid. 

Any thoughts about what I should do? And thank you for your help, Brian!

**Reply by brianpetro** [reply]
It looks like the embeddings.json file isn't being created or loaded. Check out this https://github.com/brianpetro/obsidian-smart-connections/issues/244#issuecomment-1603458823

🌴

**Reply by perryfr** [reply]
Hmm, I wasn't able to get the embeddings-related buttons to appear. I ended up uninstalling and reinstalling the plug-in, generating a new API key, and using the default setting (ChatGPT 3.5 turbo 16k) for the the ChatGPT model. I'm not getting any red/errors in the developer console now.

<img width="555" alt="CleanShot 2023-07-01 at 08 15 34@2x" src="https://github.com/brianpetro/obsidian-smart-connections/assets/96356433/c84af265-bc25-4c5a-b939-f0883cbcbec3">
 

In case it helps to know the steps I took prior to uninstalling the plug-in, here they are:

1. I had no notes open, turned the plug-in on and off, but that didn't work. 
2. I turned off the plug-in, closed my vault, opened the vault, and turned the plug-in back on—that didn't work. 
3. I disabled and re-enabled the plug-in—that didn't work. 
4. And I disabled the plug-in, closed and re-opened my vault, then re-enabled the plug-in—that too didn't work. 

Each time, all I got was what you see in the screenshot below.

<img width="279" alt="CleanShot 2023-07-01 at 08 18 44@2x" src="https://github.com/brianpetro/obsidian-smart-connections/assets/96356433/0883aab5-04ac-4787-b789-5c554e1723d8">

Thanks again for your assistance, Brian. Much appreciated.

**Reply by brianpetro** [reply]
@perryfr towards the bottom of the Smart Connections settings there are a couple "render log" options that you can toggle on. This will render additional logs in the developer console and may also give us additional insights into what might be happening.

Thanks for your help in solving this,
Brian 🌴
